handlers/update-supporter-plus-amount/src/updateSupporterPlusAmount.ts

```diff
@@ -0,0 +1,216 @@
+import { ValidationError } from '@modules/errors';
+import { checkDefined } from '@modules/nullAndUndefined';
+import { prettyPrint } from '@modules/prettyPrint';
+import type {
+	ProductBillingPeriod,
+	ProductCatalog,
+	ProductCurrency,
+	ProductRatePlan,
+} from '@modules/product-catalog/productCatalog';
+import { isProductCurrency } from '@modules/product-catalog/productCatalog';
+import { getAccount } from '@modules/zuora/getAccount';
+import { getSubscription } from '@modules/zuora/getSubscription';
+import type { ZuoraClient } from '@modules/zuora/zuoraClient';
+import type {
+	RatePlan,
+	RatePlanCharge,
+	ZuoraSubscription,
+} from '@modules/zuora/zuoraSchemas';
+import type { Dayjs } from 'dayjs';
+import dayjs from 'dayjs';
+import type { EmailFields } from './sendEmail';
+import { supporterPlusAmountBands } from './supporterPlusAmountBands';
+import { doUpdate } from './zuoraApi';
+
+type UpdatablePlans =
+	| 'Annual'
+	| 'Monthly'
+	| 'V1DeprecatedAnnual'
+	| 'V1DeprecatedMonthly';
+
+export type SupporterPlusData = {
+	ratePlan: RatePlan;
+	productRatePlan: ProductRatePlan<'SupporterPlus', UpdatablePlans>;
+	contributionCharge: RatePlanCharge;
+	planHasSeparateContributionCharge: boolean;
+};
+
+export const getSupporterPlusPlans = (
+	productCatalog: ProductCatalog,
+	ratePlans: RatePlan[],
+): SupporterPlusData => {
+	const supporterPlusProductData = {
+		[productCatalog.SupporterPlus.ratePlans.Monthly.id]: {
+			productRatePlan: productCatalog.SupporterPlus.ratePlans.Monthly,
+			charge:
+				productCatalog.SupporterPlus.ratePlans.Monthly.charges.Contribution,
+			planHasSeparateContributionCharge: true,
+		},
+		[productCatalog.SupporterPlus.ratePlans.Annual.id]: {
+			productRatePlan: productCatalog.SupporterPlus.ratePlans.Annual,
+			charge:
+				productCatalog.SupporterPlus.ratePlans.Annual.charges.Contribution,
+			planHasSeparateContributionCharge: true,
+		},
+		[productCatalog.SupporterPlus.ratePlans.V1DeprecatedMonthly.id]: {
+			productRatePlan:
+				productCatalog.SupporterPlus.ratePlans.V1DeprecatedMonthly,
+			charge:
+				productCatalog.SupporterPlus.ratePlans.V1DeprecatedMonthly.charges
+					.Subscription,
+			planHasSeparateContributionCharge: false,
+		},
+		[productCatalog.SupporterPlus.ratePlans.V1DeprecatedAnnual.id]: {
+			productRatePlan:
+				productCatalog.SupporterPlus.ratePlans.V1DeprecatedAnnual,
+			charge:
+				productCatalog.SupporterPlus.ratePlans.V1DeprecatedAnnual.charges
+					.Subscription,
+			planHasSeparateContributionCharge: false,
+		},
+	};
+
+	const productRatePlans = ratePlans
```

Comment:
something like this..
```suggestion
	const updatableChargeData = ratePlans
```